Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1125,208 +1125,44 @@ open class NavigationMapView: UIView {
-            
-            var coordinate: CLLocationCoordinate2D?
-            
-            // Obtain a polyline of the set of steps. We'll look for a good spot along this line to
-            // place the annotation.
-            // We will consider a good spot to be somewhere near the middle of the line, making sure
-            // that the coordinate is visible on-screen.
-            if let continuousLine = visibleAlternateSteps.continuousShape(),
-                continuousLine.coordinates.count > 0 {
-                coordinate = continuousLine.coordinates[0]
-                
-                // Pick a coordinate using some randomness in order to give visual variety.
-                // Take care to snap that coordinate to one that lays on the original route line.
-                // If the chosen snapped coordinate is not visible on the screen, then we walk back
-                // along the route coordinates looking for one that is.
-                // If none of the earlier points are on screen then we walk forward along the route
-                // coordinates until we find one that is.
-                if let distance = continuousLine.distance(),
-                    let sampleCoordinate = continuousLine.indexedCoordinateFromStart(distance: distance * CLLocationDistance.random(in: 0.3...0.8))?.coordinate,
-                    let routeShape = route.shape,
-                    let snappedCoordinate = routeShape.closestCoordinate(to: sampleCoordinate) {
-                    var foundOnscreenCoordinate = false
-                    var firstOnscreenCoordinate = snappedCoordinate.coordinate
-                    for indexedCoordinate in routeShape.coordinates.prefix(through: snappedCoordinate.index).reversed() {
-                        if visibleBoundingBox.contains(indexedCoordinate) {
-                            firstOnscreenCoordinate = indexedCoordinate
-                            foundOnscreenCoordinate = true
-                            break
-                        }
-                    }
-                    
-                    if foundOnscreenCoordinate {
-                        // We found a point that is both on the route and on-screen.
-                        coordinate = firstOnscreenCoordinate
-                    } else {
-                        // We didn't find a previous point that is on-screen so we'll move forward
-                        // through the coordinates looking for one.
-                        for indexedCoordinate in routeShape.coordinates.suffix(from: snappedCoordinate.index) {
-                            if visibleBoundingBox.contains(indexedCoordinate) {
-                                firstOnscreenCoordinate = indexedCoordinate
-                                break
-                            }
-                        }
-                        coordinate = firstOnscreenCoordinate
-                    }
-                }
+    public var showsIntersectionSignals: Bool = false {
```

Comment:
This feature flag should be a property of NavigationViewController and CarPlayNavigationViewController, similar to `routeLineTracksTraversal`. The map view should only know how to show and hide these annotations.

Reply:
This property has been moved to `NavigationViewController` and `CarPlayNavigationViewController`  for active navigation session only.